ApplicationLibCode/ProjectDataModel/Flow/RimWellAllocationPlot.cpp

```diff
@@ -219,6 +221,57 @@ void RimWellAllocationPlot::deleteViewWidget()
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimWellAllocationPlot::setCase( RimEclipseResultCase* eclipseCase )
+{
+    bool emptyPreviousCase = !m_case;
+
+    m_case = eclipseCase;
+
+    if ( m_case )
+    {
+        m_flowDiagSolution = m_case->defaultFlowDiagSolution();
+    }
+    else
+    {
+        m_flowDiagSolution = nullptr;
+        m_timeStep         = 0;
+    }
+
+    if ( !m_case )
+        m_timeStep = 0;
+    else
+    {
+        if ( emptyPreviousCase )
+        {
+            m_timeStep = (int)( m_case->timeStepDates().size() - 1 );
+        }
+
+        m_timeStep = std::min( m_timeStep(), ( (int)m_case->timeStepDates().size() ) - 1 );
+    }
```

Comment:
Combine these two `ifs`? Has the same condition.

Reply:
Agree, fixed